authors/apps/comments/views.py

```diff
@@ -0,0 +1,211 @@
+        self.get_author_profile(request)
+        if not self.is_author:
+            return Response({"response": "You can only delete a comment you authored",
+                            "status_message": "Failed: Access denied"},
+                            status=status.HTTP_403_FORBIDDEN)
+        self.comment.delete()
+        return Response({
+            "status_message": "Successfully deleted comment: {}".format(pk)},
+            status=status.HTTP_200_OK)
+
+
+class CommentReplyApiView (GenericAPIView):
+    """
+    This ApiView class handles the addition of a reply to a comment
+    and retrieving of all replies to that comment.
+    """
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentReplySerializer
+    renderer_classes = (JSONRenderer,)
+
+    def get_required_objects(self, request, comment_pk):
+        self.comment = get_object_or_404(Comment, pk=comment_pk)
+        self.data = request.data.get(
+            'reply') if 'reply' in request.data else request.data
+
+    def get_author_profile(self, request):
+        self.author = Profile.objects.get(user=request.user)
+
+    def get(self, request, comment_pk):
+        """Retrieve all replies on an comment"""
+        self.get_required_objects(request, comment_pk)
+        replies = CommentReply.objects.filter(comment=self.comment)
+        serialized_data = self.serializer_class(replies, many=True)
+        return Response({
+            "replies": serialized_data.data,
+            "repliesCount": replies.count(),
+            "status_message": "Successfully returned replies to comment: {}".format(comment_pk)},
+            status=status.HTTP_200_OK
+        )
+
+    def post(self, request, comment_pk):
+        """Make a reply to a comment"""
+        self.get_required_objects(request, comment_pk)
+        self.get_author_profile(request)
+        serializer = self.serializer_class(
+            data=self.data
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save(
+            author=self.author,
+            comment=self.comment
+        )
+        return Response({"reply": serializer.data,
+            "status_message": "Successfully posted reply to comment: {}".format(comment_pk)},
+            status=status.HTTP_201_CREATED)
+
+
+class CommentReplyDetailApiView (GenericAPIView):
+    """
+    The CommentReplyDetailApiView handles the retreiving, modification and 
+    deletion of a single reply.
+    """
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (JSONRenderer,)
+
+    def get_required_objects(self, request, pk):
+        self.reply = get_object_or_404(CommentReply, pk=pk)
+        self.data = request.data.get(
+            'reply') if 'reply' in request.data else request.data
+
+    def get_author_profile(self, request):        
+        self.requester = Profile.objects.get(user=request.user)
```

Comment:
@pbkabali I think you should use either a get_object_or_404 or a try and except just incase a Profile does not exist a get might throw an error if that profile does not exist

Reply:
@oma0256 I get your point because the get endpoints do not need authentication. But also this function is not called on any get endpoint. So I think it should be fine